moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -246,7 +255,7 @@ class Robot
    */
   bool jointInterpolation(const std::unique_ptr<TrajectoryPoint> & traj_point,
            std::vector<trajectory_msgs::JointTrajectoryPoint> & points,
-           const unsigned int num_steps);
+           const unsigned int num_steps, bool collision_check = false);
```

Comment:
Are you sure you want to check for collisions here?  Isn't this an internal method?  It would be more straight-forward to do this check once, in the plan/execute method.